lib/extensions/ar_migration.rb

```diff
@@ -0,0 +1,84 @@
+module ArPglogicalMigration
+  module PglogicalMigrationHelper
+    def self.migrations_column_present?
+      ActiveRecord::Base.connection.columns("miq_regions").any? { |c| c.name == "migrations_ran" }
+    end
+
+    def self.update_local_migrations_ran(version, direction)
+      return unless migrations_column_present?
+      return unless (region = MiqRegion.my_region)
+
+      new_migrations = ActiveRecord::SchemaMigration.normalized_versions
+      new_migrations << version if direction == :up
+      migrations_value = ActiveRecord::Base.connection.quote(PG::TextEncoder::Array.new.encode(new_migrations))
+
+      ActiveRecord::Base.connection.exec_query(<<~SQL)
+        UPDATE miq_regions
+        SET migrations_ran = #{migrations_value}
+        WHERE id = #{region.id}
+      SQL
+    end
+  end
+
+  class RemoteRegionMigrationWatcher
```

Comment:
Okay, this might be slightly better than `RemoteRegionThing` :laughing: https://github.com/carbonin/manageiq/pull/4#discussion_r219559690

Reply:
`Stuff` and `Thing`... the best placeholder names that gets you to the point of understanding what it actually does... and hopefully no reviewer will actually merge it until it's renamed.

Reply:
> hopefully no reviewer will actually merge it until it's renamed

Hopefully ... but not always

```
$ git grep "it \"works\""
spec/lib/extensions/ar_taggable_spec.rb:125:    it "works" do
spec/lib/extensions/ar_taggable_spec.rb:143:    it "works" do
spec/lib/rbac_spec.rb:30:      it "works" do
spec/lib/rbac_spec.rb:61:        it "works" do
spec/lib/rbac_spec.rb:75:        it "works" do
spec/models/authentication_spec.rb:38:    it "works" do
spec/models/category_spec.rb:3:    it "works" do
spec/models/miq_approval_spec.rb:16:    it "works" do
spec/models/mixins/authentication_mixin_spec.rb:144:    it "works" do
spec/models/mixins/authentication_mixin_spec.rb:205:          it "works" do
spec/models/vm_or_template_spec.rb:1096:    it "works" do
spec/models/vm_or_template_spec.rb:1106:    it "works" do
spec/models/vm_or_template_spec.rb:1116:    it "works" do
```

Reply:
Don't forget:
```
$ git grep "it \"stuff\""
spec/lib/vmdb/loggers/container_logger_spec.rb:2:  it "stuff" do
```

Reply:
Awesome

Reply:
> Hopefully ... but not always

I bet every one of those are @bdunne 😆

Reply:
> I bet every one of those are @bdunne 

I think only `it "stuff"` is me actually